cvmfs/sync_item.cc

```diff
@@ -263,7 +263,18 @@ void SyncItem::CheckMarkerFiles() {
 }
 
 void SyncItem::CheckCatalogMarker() {
-  has_catalog_marker_ = FileExists(GetUnionPath() + "/.cvmfscatalog");
+  std::string path(GetUnionPath() + "/.cvmfscatalog");
+  EntryStat stat;
```

Comment:
@siscia has found that the structure `EntryStat` should also check the `error_code` just as it is done with the  `obtained` in `assert(obtained)`.

Reply:
She is referring to this code here: https://github.com/cvmfs/cvmfs/blob/e3fa33ad1543076bf5d8f9322e0c34522330d42b/cvmfs/sync_item.h#L230-L250

Where we don't check if there was an error during the stat call.